storage/database/src/backend/heed/storable.rs

```diff
@@ -16,7 +16,19 @@ pub(super) struct StorableHeed<T>(PhantomData<T>)
 where
     T: Storable + ?Sized;
 
-//---------------------------------------------------------------------------------------------------- BytesDecode
+//---------------------------------------------------------------------------------------------------- Key
+// If `Key` is also implemented, this can act as the comparison function.
+impl<T> heed::Comparator for StorableHeed<T>
+where
+    T: Key,
+{
+    #[inline]
+    fn compare(a: &[u8], b: &[u8]) -> Ordering {
+        <T as Key>::compare(a, b)
+    }
+}
```

Comment:
## Key 3/4
This implementation on our compatibility wrapper type. It just forwards our `Key::compare` to `heed`'s compare function.

Now the database table will use `Key::compare` to compare keys.